modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/outgoing/OutgoingSnapshot.java

```diff
@@ -71,18 +76,23 @@ public class OutgoingSnapshot {
 
     private final ReusableLockLockup mvOperationsLockup = new ReusableLockLockup(mvOperationsLock);
 
+    /**
+     * Snapshot metadata taken on snapshot scope freezing.
+     */
+    private SnapshotMeta meta;
+
     /**
      * {@link RowId}s for which the corresponding rows were sent out of order (relative to the order in which this
      * snapshot sends rows), hence they must be skipped when sending rows normally.
      */
     private final Set<RowId> rowIdsToSkip = new ConcurrentHashSet<>();
 
-    // TODO: IGNITE-17935 - manage queue size
+    // TODO: IGNITE-18018 - manage queue size
     /**
      * Rows that need to be sent out of order (relative to the order in which this snapshot sends rows).
      * Versions inside rows are in oldest-to-newest order.
      */
-    private final Queue<SnapshotMvDataResponse.ResponseEntry> outOfOrderMvData = new LinkedList<>();
+    private final Queue<SnapshotMvDataResponse.ResponseEntry> outOfOrderMvData = new ArrayDeque<>();
```

Comment:
I don't see where it is protected when calling method `java.util.Queue#poll`.

Reply:
`try (AutoLockup ignored = acquireMvLock()) {` in `handleSnapshotMvDataRequest()`

Reply:
Please add assertion